service/history/api/recordworkflowtaskstarted/api.go

```diff
@@ -151,6 +151,18 @@ func Invoke(
 				mutableState.ClearStickyTaskQueue()
 			}
 
+			// TODO (carly): Confirm that this handles obsolete task starts correctly
+			// Versioning reschedules tasks when there are changes to effective deployment or behavior, which
+			// can result in obsolete dispatch requests when the task that was originally scheduled tries to
+			// start after it has already been rescheduled.
+			// Even if the current task queue is not sticky, we want to reject these start requests.
+			pollerVersionCaps := req.GetPollRequest().GetWorkerVersionCapabilities()
```

Comment:
This if should go inside the if we have in line 168. We should not drop tasks if their source deployment match because they are redirected by matching. for that we need to check TaskRedirectInfo.schedule_time_workflow_deployment` which is added in [here](https://github.com/temporalio/temporal/pull/6874/files). For now you can remove this check from this PR.

Reply:
removed